prime-dotnet-webapi/Services/DefaultEnrolleeService.cs

```diff
@@ -318,109 +318,89 @@ public Task<EnrolmentStatus> CreateEnrolmentStatusAsync(int enrolleeId, Status s
                         };
-                        break;
-                    case Status.DECLINED_CODE:
-                        await SetAllPharmaNetStatusesFalseAsync(enrolleeId);
-                        createdEnrolmentStatus.PharmaNetStatus = true;
-                        break;
-                    case Status.ACCEPTED_TOS_CODE:
-                        await SetAllPharmaNetStatusesFalseAsync(enrolleeId);
-                        enrollee.LicensePlate = this.GenerateLicensePlate();
-                        createdEnrolmentStatus.PharmaNetStatus = true;
-                        break;
-                    case Status.DECLINED_TOS_CODE:
-                        await SetAllPharmaNetStatusesFalseAsync(enrolleeId);
-                        createdEnrolmentStatus.PharmaNetStatus = true;
-                        break;
-                }
+                        adjudicatedEnrolmentStatus.AddStatusReason(StatusReason.AUTOMATIC_CODE);
 
-                if (Status.ACCEPTED_TOS_CODE.Equals(status?.Code))
-                {
-                    enrollee.LicensePlate = this.GenerateLicensePlate();
-                }
+                        enrollee.EnrolmentStatuses.Add(adjudicatedEnrolmentStatus);
+                        // Flip to the object that will get returned
+                        createdEnrolmentStatus = adjudicatedEnrolmentStatus;
+                    }
+                    break;
 
-                var created = await _context.SaveChangesAsync();
+                case Status.APPROVED_CODE:
+                    createdEnrolmentStatus.AddStatusReason(StatusReason.MANUAL_CODE);
+                    break;
 
-                if (created < 1)
-                {
-                    throw new InvalidOperationException("Could not create enrolment status.");
-                }
+                case Status.DECLINED_CODE:
+                    await SetAllPharmaNetStatusesFalseAsync(enrolleeId);
+                    createdEnrolmentStatus.PharmaNetStatus = true;
+                    break;
 
-                // Enrollee just left manual adjudication, inform the enrollee
-                if (currentStatus.Code == Status.SUBMITTED_CODE)
-                {
-                    _emailService.SendReminderEmail(enrollee);
-                }
+                case Status.ACCEPTED_TOS_CODE:
+                    await SetAllPharmaNetStatusesFalseAsync(enrolleeId);
+                    enrollee.LicensePlate = this.GenerateLicensePlate();
+                    createdEnrolmentStatus.PharmaNetStatus = true;
+                    break;
+
+                case Status.DECLINED_TOS_CODE:
+                    await SetAllPharmaNetStatusesFalseAsync(enrolleeId);
+                    createdEnrolmentStatus.PharmaNetStatus = true;
+                    break;
+            }
 
-                return createdEnrolmentStatus;
+            var created = await _context.SaveChangesAsync();
+            if (created < 1)
```

Comment:
This reads weird doesn't it?What is greater than 1?

Reply:
this returns the number of records created when saving the database. It's existing code that has moved in the file